makerepropkg.in

```diff
@@ -172,23 +190,43 @@ BUILDTOOLVER="${buildinfo[buildtoolver]}"
 PKGEXT=${splitpkgs[0]#${splitpkgs[0]%.pkg.tar*}}
 
 # nuke and restore reproducible testenv
-for copy in "${buildroot}"/*/; do
+namespace="$buildroot/$chroot"
+lock 9 "${namespace}.lock" "Locking chroot namespace '%s'" "${namespace}"
+for copy in "${namespace}"/*/; do
     [[ -d ${copy} ]] || continue
     subvolume_delete_recursive "${copy}"
 done
-rm -rf --one-file-system "${buildroot}"
-(umask 0022; mkdir -p "${buildroot}")
+rm -rf --one-file-system "${namespace}"
+(umask 0022; mkdir -p "${namespace}")
 
 for fname in "${installed[@]}"; do
     if ! allpkgfiles+=("$(get_pkgfile "${fname}")"); then
         error "failed to retrieve ${fname}"
         exit 1
     fi
 done
-printf '%s\n' "${allpkgfiles[@]}" | mkarchroot -M @pkgdatadir@/makepkg-x86_64.conf -U "${archroot_args[@]}" "${buildroot}"/root - || exit 1
+
+trap 'rm -rf $TEMPDIR' EXIT INT TERM QUIT
+TEMPDIR=$(mktemp -d --tmpdir makerepropkg.XXXXXXXXXX)
+
+makepkg_conf="${TEMPDIR}/makepkg.conf"
+# anything before buildtool support is pinned to last none buildtool devtools
+if [[ -z "${BUILDTOOL}" ]]; then
+    get_makepkg_conf "devtools-20210202-3-any" "${makepkg_conf}"
+# prefere to assume devtools up until matching makepkg version so repository packages remain reproducible
+elif [[ "${BUILDTOOL}" = makepkg ]] && (( $(vercmp "${BUILDTOOLVER}" 6.0.1) <= 0 )); then
+    get_makepkg_conf "devtools-20210202-3-any" "${makepkg_conf}"
+# all devtools builds
+elif [[ "${BUILDTOOL}" = devtools ]]; then
+    get_makepkg_conf "${BUILDTOOL}-${BUILDTOOLVER}" "${makepkg_conf}"
+# fallback to current makepkg.conf
+else
+    makepkg_conf=@pkgdatadir@/makepkg-x86_64.conf
```

Comment:
This probably warrants a `msg2 falling back to default makepkg-x86_64.conf?

Or maybe to reflect my issue, what if I set an invalid buildtool, I'll never notice that it's wrong, so should it be printed explicitly?